### PR TITLE
add some more scaladoc comments

### PR DESCRIPTION
**Pull request by stew**

### CODE REVIEW

File: core/src/main/scala/cats/MonadFilter.scala [hunk 1]
```diff
@@ -2,6 +2,13 @@ package cats
 
 import simulacrum._
 
+/**
+ * a Monad equipped with an additional method which allows us to
+ * create an "Empty" value fro the Monad (for whatever "empty" makes
```
Comment (ceedubs) [inline core/src/main/scala/cats/MonadFilter.scala]: s/fro/for/

File: core/src/main/scala/cats/MonadFilter.scala [hunk 2]
```diff
@@ -2,6 +2,13 @@ package cats
...
+ * sense for that particular monad. This is of particular interest to
```
Comment (ceedubs) [inline core/src/main/scala/cats/MonadFilter.scala]: My mental parser is demanding an end parenthesis.

File: core/src/main/scala/cats/MonadFilter.scala [hunk 3]
```diff
@@ -2,6 +2,13 @@ package cats
...
+ * us since it allows us to add a `filter` method to a Monad, which is
+ * used when patter matching or using guards in for comprehensions.
```
Comment (ceedubs) [inline core/src/main/scala/cats/MonadFilter.scala]: s/patter/pattern/

File: core/src/main/scala/cats/Show.scala [hunk 4]
```diff
@@ -3,7 +3,11 @@ package cats
 import cats.functor.Contravariant
 
 /**
- * A typeclass to provide textual representation
+ * A typeclass to provide textual representation. It is meant to be a
+ * better "toString". Whereas toString exists for any Object,
+ * regardless of whether or not the creator of the class explicitly
+ * made a toString method, a Show instance will only exist if someone
+b * explicitly provided one.
```
Comment (ceedubs) [inline core/src/main/scala/cats/Show.scala]: stray `b`

File: core/src/main/scala/cats/Traverse.scala [hunk 5]
```diff
@@ -16,14 +16,29 @@ import simulacrum._
  */
 @typeclass trait Traverse[F[_]] extends Functor[F] with Foldable[F] { self =>
 
+  /**
+   * given a function which returns a G effect, thread this effect
+   * through the running of this function on all the values in F,
+   * returning an F[A] in a F context
```
Comment (ceedubs) [inline core/src/main/scala/cats/Traverse.scala]: Do you mean returning an `F[B]` in a `G` context?

**Comment by ceedubs** [comment]
Thanks, @stew. Great stuff. I made some nitpicky typo comments, but I definitely look forward to this being merged after those are fixed.

**Comment by stew** [comment]
(and that one (F[B] in a G context) was fixed by the last PR)

**Comment by non** [comment]
Looks good to me :+1:

**Pull request merged**